### ISSUE TITLE
Is it better that output seq (& qual) not be folded for FASTQ?

### ISSUE THREAD

**Issue by omegahh** [cartoonist/kseqpp#6]
https://github.com/cartoonist/kseqpp/blob/558de3086e66dbbdbe1c4bfcc8d0f5abff59b19b/include/kseq%2B%2B/kseq%2B%2B.hpp#L291

**Comment by cartoonist** [comment]
Sorry, I did not get what you mean by being folded. You mean wrapped?

**Comment by cartoonist** [comment]
I added an option to disable line wrapping by calling `set_nowrapping` method before writing any record. Since this is re-implementation of `kseq` library, I did not change the default behaviour.

**Comment by omegahh** [comment]
Yes, I mean 'wrapped'. For most FASTQ files, The seq line and qual line are not wrapped. As far as I know, the wrapped FASTQ is not accepted by many downstream softwares, for example: bowtie2.
So my suggestion is, in default manner, the FASTA file should be wrapped while the FASTQ file should not be wrapped.

**Comment by cartoonist** [comment]
That makes sense. I am going to do the following then:
- add a parameter to `SeqStreamOut` class to specify the underlying file format (accepting `format::fasta`, or `format::fastq` values in `klibpp` namespace). Actually `KStreamIn` class which is the base class of `SeqStreamOut` has already this option which just needs to be forwarded.
- if the format is explicitly set to `format::fastq`, no-wrapping would be the default behaviour, otherwise it wraps at 60bps.
- just for the record, `puts` no longer get `wrap` parameter and wrapping is determined by the `wraplen` value. If it is zero, it means no wrap.

**Comment by cartoonist** [comment]
Please check out `develop` branch (sorry for the forced push!), let me know if it works.

**Comment by omegahh** [comment]
Sorry for delayed reply. I am on the vocation due to Chinese national day.

**Comment by cartoonist** [comment]
No problem, feel free to reopen the issue if the problem persists.
